lib/ruby_lsp/requests/document_symbol.rb

```diff
@@ -106,8 +106,36 @@ def visit_class(node)
         @stack.pop
       end
 
+      sig { override.params(node: SyntaxTree::MethodAddBlock).void }
+      def visit_method_add_block(node)
```

Comment:
I am not sure what this is matching, can you clarify why we need this in addition to what `visit_commands` does?

If we still do need this, I think we should share a lot of this logic by calling out to `parse_test` anyway.